client/lib/cgroupslib/partition_linux.go

```diff
@@ -49,50 +53,60 @@ func NewPartition(cores *idset.Set[hw.CoreID]) Partition {
 }
 
 type partition struct {
+	log         hclog.Logger
 	sharePath   string
 	reservePath string
+	usableCores *idset.Set[hw.CoreID]
 
 	lock    sync.Mutex
 	share   *idset.Set[hw.CoreID]
 	reserve *idset.Set[hw.CoreID]
 }
 
 func (p *partition) Restore(cores *idset.Set[hw.CoreID]) {
+
 	p.lock.Lock()
 	defer p.lock.Unlock()
 
 	p.share.RemoveSet(cores)
```

Comment:
Noting for other reviewers, because I had to check, that removing from the set is safe even if the cores don't exist in the set: https://go.dev/play/p/3iZeKQThSaV